packages/components/src/tools-panel/tools-panel-item/hook.ts

```diff
@@ -54,7 +54,11 @@ export function useToolsPanelItem(
 			} );
 		}
 
-		return () => deregisterPanelItem( label );
+		return () => {
+			if ( currentPanelId === panelId ) {
+				deregisterPanelItem( label );
+			}
+		}
```

Comment:
Looks like the linter is complaining about a missing semi-colon here 🙂
```suggestion
		};
```

Reply:
Moving too fast 🙈